plugins/modules/purefa_snmp.py

```diff
@@ -162,108 +174,122 @@ def test_manager(module, array):
 def update_manager(module, array):
     """Update SNMP Manager"""
     changed = False
-    try:
-        mgr = array.get_snmp_manager(module.params["name"])
-    except Exception:
+    res = array.get_snmp_managers(names=module.params["name"])
+    if res.status_code != 200:
         module.fail_json(
             msg="Failed to get configuration for SNMP manager {0}.".format(
                 module.params["name"]
             )
         )
-    if mgr["version"] != module.params["version"]:
+    else:
+        mgr = list(res.items)[0]
+    if mgr.version != module.params["version"]:
         module.fail_json(msg="Changing an SNMP managers version is not supported.")
     elif module.params["version"] == "v2c":
         changed = True
         if not module.check_mode:
-            try:
-                array.set_snmp_manager(
-                    module.params["name"],
-                    community=module.params["community"],
+            res = array.patch_snmp_managers(
+                names=module.params["name"],
+                version=module.params["version"],
```

Comment:
"version" should be under SnmpManagerPatch, pls move.
With this fix, we are good to merge.